[4.0] Brought checkbox and its label in the same line

saumyasarkar11 opened the pull request:
Pull Request to bring label "Remember me" and its checkbox in the same line .

### Actual result BEFORE applying this Pull Request

![Screenshot from 2021-03-23 15-25-15](https://user-images.githubusercontent.com/76894046/112128322-546e7580-8bec-11eb-8b77-e71b96f404ef.png)


### Expected result AFTER applying this Pull Request

![Screenshot from 2021-03-23 15-24-04](https://user-images.githubusercontent.com/76894046/112128133-1cffc900-8bec-11eb-95d1-667ae82c80d6.png)

### Documentation Changes Required

None

saumyasarkar11 committed:
Brought checkbox and its label in the same line

drmenzelit commented on components/com_users/tmpl/login/default_login.php:
```diff
@@ -60,14 +60,12 @@
+					
+					<input id="remember" type="checkbox" name="remember" class="inputbox" value="yes">
+					
+					<label for="remember" style="margin-left:3px;">
```
No inline style, please

brianteeman commented:
dont we have a class form-check-inline for this?

drmenzelit commented:
https://getbootstrap.com/docs/5.0/forms/checks-radios/#checks

saumyasarkar11 commented:
Working on it

saumyasarkar11 committed:
Addition of classes in tags

Addition of classes in tags as per Bootstrap 5 and removal of internal styling in label

saumyasarkar11 commented:
Please go through the new commit @brianteeman @drmenzelit

brianteeman commented on components/com_users/tmpl/login/default_login.php:
```diff
@@ -60,14 +60,14 @@
-					<div class="controls">
-						<input id="remember" type="checkbox" name="remember" class="inputbox" value="yes">
-					</div>
+					
```
not 100% certain so please check but you can probably remove control-group

richard67 commented:
@saumyasarkar11 At the bottom of your PR you can see that CI checks by drone are failing:

![2021-03-23_01](https://user-images.githubusercontent.com/7413183/112146033-5ed44380-8bdb-11eb-9e22-7fa8588d34fe.png)

If you follow the "Details" link you come to a page where you can see the tests at the left hand side. The one called PHPCS failed (red colour). If you then click on that PHPCS you will see the log file:

https://ci.joomla.org/joomla/joomla-cms/41089/1/6

This one then report code style errors:

```
FILE: /********/src/components/com_users/tmpl/login/default_login.php
----------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
 65 | ERROR | [x] Tabs must be used to indent lines; spaces are not
    |       |     allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 66 | ERROR | [x] Tabs must be used to indent lines; spaces are not
    |       |     allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 67 | ERROR | [x] Tabs must be used to indent lines; spaces are not
    |       |     allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
 68 | ERROR | [x] Tabs must be used to indent lines; spaces are not
    |       |     allowed
    |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
```

If you fix these code style errors and push the changes, the CI checks will start again, and then they should pass.

Code style is important for a project's maintainability.

The PHPCS is important because if that fails, the automated system tests are not run, which would show other possible errors.

saumyasarkar11 committed:
Fixed code style errors

saumyasarkar11 commented:
I think I need to go a bit deeper. Working on it

richard67 commented:
Maybe 2 helpful links regarding code style:

- Find an instruction on how to check and fix PHP code with our code style rules for Joomla 4 here: https://docs.joomla.org/Joomla_CodeSniffer
- Find a description of our code style rules also for other languages here for Joomla 3: https://developer.joomla.org/coding-standards/basic-guidelines.html ... these basically also apply to Joomla 4, just that in J4 we have scss and not less.

saumyasarkar11 committed:
Re: Coding style implemented

drmenzelit commented:
Please check the comment from Brian: https://github.com/joomla/joomla-cms/pull/32821#discussion_r599482289

saumyasarkar11 commented:
> Please check the comment from Brian: [#32821 (comment)](https://github.com/joomla/joomla-cms/pull/32821#discussion_r599482289)

I've removed that additional div. He had commented on the outdated version

richard67 commented:
> I've removed that additional div. He had commented on the outdated version

@saumyasarkar11 I'm not sure. I thin Brian meant this one:

https://github.com/joomla/joomla-cms/pull/32821/files#diff-4b87fc46e3b0b4efde5b14360a34efca58df76b798e0956948ce45a3e755da0fR62

And as far as I understood he was not sure about if it can be removed or not, but it could be.

If possible, could you check that?

drmenzelit commented:
@saumyasarkar11 it is a simple check: look into the scss/css files from Cassiopeia template and check if the control-group class is still defined.

saumyasarkar11 committed:
Removed unnecessary class on line 62

saumyasarkar11 commented:
> @saumyasarkar11 it is a simple check: look into the scss/css files from Cassiopeia template and check if the control-group class is still defined.

It is still defined but it is as follows:
.control-group {
  margin: 1em 0;
}
May be we can remove it.

drmenzelit commented:
No, the class is still been used in several views, e.g. `/components/com_content/tmpl/form/edit.php`
Related issue: https://github.com/joomla/joomla-cms/issues/31218

saumyasarkar11 commented:
> No, the class is still been used in several views, e.g. `/components/com_content/tmpl/form/edit.php`
> Related issue: #31218

Putting it back in

drmenzelit commented:
Not necessary, your PR looks good now.

drmenzelit commented:
I have tested this item :white_check_mark: successfully on dd33b423df6a851c93240b734e8f05c1ebf9a426<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32821">issues.joomla.org/tracker/joomla-cms/32821</a>.</sub>

brianteeman commented on components/com_users/tmpl/login/default_login.php:
```diff
@@ -59,15 +59,13 @@
-				<div  class="com-users-login__remember control-group">
-					<div class="control-label">
-						<label for="remember">
+				<div  class="com-users-login__remember">					
```
```suggestion
				<div class="com-users-login__remember">					
```

please remove extra space

ChristineWk replied:
> 
> 
> please remove extra space

was not seen before (under Issue)

ChristineWk commented:
I have tested this item :white_check_mark: successfully on dd33b423df6a851c93240b734e8f05c1ebf9a426<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32821">issues.joomla.org/tracker/joomla-cms/32821</a>.</sub>

ChristineWk commented:
https://github.com/joomla/joomla-cms/pull/32821#discussion_r599796595
was not seen before (under Issue)

saumyasarkar11 committed:
Removed extra space on line 62

Co-authored-by: Brian Teeman <brian@teeman.net>

saumyasarkar11 commented:
extra space removed. Please review

> I have tested this item  successfully on [dd33b42](https://github.com/joomla/joomla-cms/commit/dd33b423df6a851c93240b734e8f05c1ebf9a426)
> 
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/32821](https://issues.joomla.org/tracker/joomla-cms/32821).

Extra space removed. Please review

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32821">issues.joomla.org/tracker/joomla-cms/32821</a>.</sub>

saumyasarkar11 commented:
> RTC
> 
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/32821](https://issues.joomla.org/tracker/joomla-cms/32821).

It says Failure: Number of issues exceeds sum threshold (1/0). What does it mean? @Quy

richard67 commented:
> > RTC
> > This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/32821](https://issues.joomla.org/tracker/joomla-cms/32821).
> 

"RTC" means "ready to commit" and is the status which a PR gets here when it has 2 successful human tests by someone else than the author.

> It says Failure: Number of issues exceeds sum threshold (1/0). What does it mean? @Quy

The failure in the "analysis4x" step drone CI isn't related to this PR. It's a security check (using RIPS) which sometimes might produce false alarms until adjusted, clarification is already ongoing.
